pkg/machine/applehv/machine.go

```diff
@@ -186,9 +187,40 @@ func (m *MacMachine) writeIgnitionConfigFile(opts machine.InitOptions, key strin
 
 func (m *MacMachine) Init(opts machine.InitOptions) (bool, error) {
 	var (
-		key          string
-		virtiofsMnts []machine.VirtIoFs
+		key                string
+		virtiofsMnts       []machine.VirtIoFs
+		cleanCallbackFuncs []func() error
+		err                error
 	)
+
+	// cleanup half-baked files if init fails at any point, or if init is sigterm'd
+	cleanCallback := func() {
+		for _, clean := range cleanCallbackFuncs {
+			if err := clean(); err != nil {
+				logrus.Error(err)
+			}
+		}
+	}
+	defer func() {
+		if err != nil {
+			cleanCallback()
+		}
+	}()
+
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+	go func() {
+		_, ok := <-c
+		if !ok {
+			return
+		}
+		cleanCallback()
```

Comment:
I think there is a data race here where a signal interrupt occurs mid function and cleanCallbackFuncs is/was mutating. You could wrap the appends and the the coordinating cleanup call inside an object with a mutex as one approach for a solution.